c/datasetjson.c

```diff
@@ -908,6 +916,44 @@ static void respondWithDYNALLOCError(HttpResponse *response,
 
 }
 
+static void getDYNALLOCErrorCodeAndMsg(int rc, int sysRC, int sysRSN,
+                                       const DynallocDatasetName *dsn,
+                                       const DynallocMemberName *member,
+                                       const char *site,
+                                       char* responseMessage,
+                                       int* responseCode) {
+
+  if (rc ==  RC_DYNALLOC_SVC99_FAILED && sysRC == 4) {
```

Comment:
this is code duplication with above, let's try to put this into a common function, because as we discover more reasons to print out, we'll want to add them in both places and it could be a maintenance nightmare later.

Reply:
perhaps this is just the difference of whether to printf or to respond with message, depending on which object got passed in.

Reply:
get rid of 
static void respondWithDYNALLOCError(HttpResponse *response,
                                     int rc, int sysRC, int sysRSN,
                                     const DynallocDatasetName *dsn,
                                     const DynallocMemberName *member,
                                     const char *site) {

just have 

static void getDYNALLOCErrorCodeAndMsg(int rc, int sysRC, int sysRSN,
                                       const DynallocDatasetName *dsn,
                                       const DynallocMemberName *member,
                                       const char *site,
                                       char* responseMessage,
                                       int* responseCode) {

then, users of respondWithDYNALLOCError could just receive `responseMessage` and `responseCode` and do `respondWithMessage` by themselves.